extensions/blocks/social-previews/constants.js

```diff
@@ -1,24 +1,42 @@
 /**
  * External dependencies
  */
+import { Fragment } from 'react';
 import { __ } from '@wordpress/i18n';
+
 import { FacebookPreview, TwitterPreview, SearchPreview } from '@automattic/social-previews';
+import { SocialServiceIcon } from '../../shared/icons';
 
 export const AVAILABLE_SERVICES = [
 	{
-		title: __( 'Google Search', 'jetpack' ),
+		title: (
+			<Fragment>
+				<SocialServiceIcon serviceName="google" />
+				{ __( 'Google Search', 'jetpack' ) }
+			</Fragment>
+		),
 		icon: 'google',
 		name: 'google',
 		preview: SearchPreview,
 	},
 	{
-		title: __( 'Facebook', 'jetpack' ),
+		title: (
+			<Fragment>
+				<SocialServiceIcon serviceName="facebook" />
+				{ __( 'Facebook', 'jetpack' ) }
+			</Fragment>
+		),
 		icon: 'facebook',
 		name: 'facebook',
 		preview: props => <FacebookPreview type="article" { ...props } />,
 	},
 	{
-		title: __( 'Twitter', 'jetpack' ),
+		title: (
+			<Fragment>
+				<SocialServiceIcon serviceName="twitter" />
+				{ __( 'Twitter', 'jetpack' ) }
+			</Fragment>
+		),
```

Comment:
While this works, I think we might want to only do this inside the `Modal` component, otherwise we won't have any other reference left of the service title in plaintext which we might sooner or later want to use. 

What do you think about generating the exact same fragment, just in modal.js, something as `AVAILABLE_SERVICES.map( … )`?

Reply:
Yeah, I explored that path initially but didn't work well for me.. But I figured it out https://github.com/Automattic/jetpack/pull/16745/commits/ca902c3ea7861834678ebec9b90674f17026b6af